addons/azureSets/azuresets.lua

```diff
@@ -68,19 +68,36 @@ windower.register_event('login', initialize)
 
 windower.register_event('job change', initialize:cond(function(job) return job == 16 end))
 
-function set_spells(spellset)
+function set_spells(spellset, setmode)
     if windower.ffxi.get_player()['main_job_id'] ~= 16 --[[and windower.ffxi.get_player()['sub_job_id'] ~= 16]] then return nil end
     if settings.spellsets[spellset] == nil then return end
-    if settings.spellsets[spellset]:equals(get_current_spellset()) then
+    if is_spellset_equipped(settings.spellsets[spellset]) then
         error(spellset..' was already equipped.')
         return
     end
 
     log('Starting to set '..spellset..'.')
-    set_spells_from_spellset(spellset, 'remove')
+    if setmode=='clearfirst' or (setmode == nil and settings.setmode == 'clearfirst') then
+        remove_all_spells(nil)
+        windower.send_command('@wait .65;lua i azuresets set_spells_from_spellset '..spellset..' add')
+    elseif setmode == 'preservetraits' or (setmode == nil and settings.setmode == 'preservetraits') or settings.setmode == nil then
+        set_spells_from_spellset(spellset, 'remove')
+    end
```

Comment:
This if-else block is a bit confusing and I think it can be done better. I assume you want to take the user-provided value, and if that is not present, use the `settings.setmode` value? If so, I would change the call-site (in the command handler) to this:
```lua
set_spells(args[1], args[2] or settings.setmode)
```

This will call the function with a second parameter, if provided, otherwise use the one from the settings table. And, judging by your code, you want it to default to 'preservertraits' if it is nil? Ideally, such a thing should never happen - settings should not be able to be `nil`, so you probably should not consider the case. The defaults can be adjusted so that they hold a value that will be used in case it's missing from the `settings.xml`. I'll add a comment to the relevant position.

Anyway, then you can rewrite the if/else-structure to this:
```lua
if setmode == 'clearfirst' then
    ...
elseif setmode == 'preservetraits' then
    ....
end
```

It will have the same effect as what you do right now and it's easier to read.